### DISCUSSION TITLE
Locale payload seems to contain all languages

### DISCUSSION THREAD

**Discussion by neverether** [i18next/next-i18next#2082] [Q&A] [answered]
I'm using `next-i18next` with NextJS 12 in production, and we've got quite a lot of localization in our codebase. On most routes we're running into data-too-large warnings from Next simply due to the localization sheets. Digging a bit deeper, I'm finding that both `en` and `ja` locales (the two locales we support) are both included in the [page].json response during hydration:

<img width="333" alt="image" src="https://user-images.githubusercontent.com/1182105/214579051-b7aa6e84-4811-4c93-b48c-3bfe8e5ed80e.png">

<img width="323" alt="image" src="https://user-images.githubusercontent.com/1182105/214579131-5a94e7c6-f6b0-45f0-b56d-a45296c77b5b.png">

(images cropped to avoid sensitive data)

The Readme clearly states only the language in use should be sent to the client but it doesn't seem to be the case: https://github.com/i18next/next-i18next#5-declaring-locale-dependencies

Before I open an issue I'd like to verify that's indeed what's happening here. Is this happening to anyone else?

EDIT:
My config is as follows:

```js
/** @type import("next").I18NConfig */
const i18n = {
  defaultLocale: "ja",
  locales: ["en", "ja"],
  localeDetection: true,
};

/** @type import("next-i18next").UserConfig */
const next18nextConfig = {
  i18n,
  defaultNS: "home",
  localePath: path.resolve("./public/locales"),
  reloadOnPrerender: isDev, // reloads the page when JSON files change – this needs to be off in production
};
```

**Comment by adrai** [comment, marked as answer]
Seems to work, also the test proofs this: https://github.com/i18next/next-i18next/blob/master/src/serverSideTranslations.test.tsx#L294
I assume your detected language is en and since your defaultLocale is ja and is always loaded as fallback, en and ja area loaded.

**Reply by neverether** [reply]
Ah you're correct, when I load a `ja` route, I'm only getting the Japanese locale data. Out of curiosity, what purpose does always loading the defaultLocale as a fallback achieve? In my experience, the fallback always ends up being the translation's key anyway.

**Reply by adrai** [reply]
Normally, the defaultLocale is your reference language... so showing that is better than showing only keys.

**Reply by neverether** [reply]
I can see how that would be preferable to keys – I personally find that a lot of users will ignore English they can't read instead of speaking up – we get more reports of missing keys when they're obviously wrong. But each to their own!

Since our team doesn't care about that fallback, I was able to set `fallbackLng: false` in my config and now I'm only getting the active locale language from the server which was my intent from the beginning to slim down the payload as much as possible.
